test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go

```diff
@@ -0,0 +1,329 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiexportendpointslice
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/core"
+	corev1alpha1 "github.com/kcp-dev/kcp/pkg/apis/core/v1alpha1"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/util/conditions"
+	topologyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/topology/v1alpha1"
+	kcpclientset "github.com/kcp-dev/kcp/pkg/client/clientset/versioned/cluster"
+	"github.com/kcp-dev/kcp/test/e2e/framework"
+)
+
+func TestAPIExportEndpointSlice(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	server := framework.SharedKcpServer(t)
+
+	// Create Organization and Workspaces
+	orgPath, _ := framework.NewOrganizationFixture(t, server)
+	var exportWs *tenancyv1alpha1.Workspace
+	exportClusterPath, exportWs := framework.NewWorkspaceFixture(t, server, orgPath, framework.WithName("export-workspace"))
+	t.Logf("Export Workspace %v", exportWs)
+	var partitionWs *tenancyv1alpha1.Workspace
+	partitionClusterPath, partitionWs := framework.NewWorkspaceFixture(t, server, orgPath, framework.WithName("partition"))
```

Comment:
could we remove `framework.WithName("partition")` ? I think that would allow for running the test multiple times without side-effects.

Reply:
ok